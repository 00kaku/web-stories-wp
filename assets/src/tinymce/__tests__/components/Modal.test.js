/*
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/**
 * External dependencies
 */
import React from 'react';
import renderer from 'react-test-renderer';
/**
 * Internal dependencies
 */
//import { forEach, isEmpty, webStoriesData } from '../../utils/globals';
/**
 * Internal dependencies
 */
import TinyMCEToggle from '../../components/controls/Toggle';
import WebStoriesModal from '../../components/Modal';

jest.mock('@wordpress/data', () => ({
  select: jest.fn(() => ({
    getCurrentView: () => 'grid',
  })),
}));

jest.mock('@wordpress/i18n', () => ({
  // eslint-disable-next-line no-unused-vars
  __: (val, domain) => val,
}));

jest.mock('@wordpress/components', () => ({
  Modal: 'Modal',
  RangeControl: 'Range',
  SelectControl: 'Select',
  Button: 'Button',
}));

jest.mock('../../utils/globals', () => ({
  forEach: () => {},
  isEmpty: () => {},
  webStoriesData: {
    views: [],
    orderlist: [],
  },
}));

jest.mock('../../components/controls/Toggle', () => 'TinyMCEToggle');

describe('TinyMCE Controls Modal', () => {
  it('modal is opened', () => {
    const props = {
      modalOpen: true,
      settings: {},
      prepareShortCode: jest.fn(),
    };

    const tree = renderer.create(<WebStoriesModal {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
