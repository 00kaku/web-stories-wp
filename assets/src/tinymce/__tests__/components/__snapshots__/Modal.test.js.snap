// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TinyMCE Controls Modal modal is opened and view is circle 1`] = `
<Modal
  className="component_web_stories_mce_model"
  closeButtonLabel="Close"
  onRequestClose={[Function]}
  shouldCloseOnClickOutside={false}
  title="Web Stories"
>
  <Select
    label="Select View Type"
    onChange={[Function]}
    options={Array []}
  />
  <TinyMCEToggle
    field="title"
  />
  <TinyMCEToggle
    field="excerpt"
  />
  <TinyMCEToggle
    field="author"
  />
  <TinyMCEToggle
    field="date"
  />
  <TinyMCEToggle
    field="image_align"
  />
  <TinyMCEToggle
    field="archive_link"
  />
  <Range
    label="Number of Stories"
    max={20}
    min={1}
    onChange={[Function]}
  />
  <Select
    label="Select Order"
    onChange={[Function]}
    options={Array []}
  />
  <div
    className="alignright"
    style={
      Object {
        "padding": "20px 0",
      }
    }
  >
    <Button
      isPrimary={true}
      onClick={[Function]}
    >
      Okay
    </Button>
    <Button
      onClick={[Function]}
    >
      Cancel
    </Button>
  </div>
</Modal>
`;

exports[`TinyMCE Controls Modal modal is opened and view is non-circle 1`] = `
<Modal
  className="component_web_stories_mce_model"
  closeButtonLabel="Close"
  onRequestClose={[Function]}
  shouldCloseOnClickOutside={false}
  title="Web Stories"
>
  <Select
    label="Select View Type"
    onChange={[Function]}
    options={Array []}
  />
  <TinyMCEToggle
    field="title"
  />
  <TinyMCEToggle
    field="excerpt"
  />
  <TinyMCEToggle
    field="author"
  />
  <TinyMCEToggle
    field="date"
  />
  <TinyMCEToggle
    field="image_align"
  />
  <TinyMCEToggle
    field="archive_link"
  />
  <Range
    label="Number of Stories"
    max={20}
    min={1}
    onChange={[Function]}
  />
  <Range
    label="Number of Columns"
    max={4}
    min={1}
    onChange={[Function]}
  />
  <Select
    label="Select Order"
    onChange={[Function]}
    options={Array []}
  />
  <div
    className="alignright"
    style={
      Object {
        "padding": "20px 0",
      }
    }
  >
    <Button
      isPrimary={true}
      onClick={[Function]}
    >
      Okay
    </Button>
    <Button
      onClick={[Function]}
    >
      Cancel
    </Button>
  </div>
</Modal>
`;
